### PR TITLE
bin(num). convert ZERO and negative decimal numbers to binary.

### LINKED ISSUE #1090: decimal_to_binary() should return identical values as bin()
https://github.com/TheAlgorithms/Python/blob/7b267e5e4f8ccb72dd58fcf0057642fd62a36bdf/conversions/decimal_to_binary.py#L4

Please change __decimal_to_binary()__ to return identical values as the Python builtin [__bin()__](https://docs.python.org/3/library/functions.html#bin).  With doctests to prove it please.

@PatOnTheBack @Corruption13

### PR DESCRIPTION
**Pull request by syedwaleedhyder**
Fixes #1090

bin(num) in decimal_to_binary.py can now convert ZERO and negative decimal numbers to binary. Consistent with built-in python bin(x) function.

### CODE REVIEW

**Comment by cclauss** [comment]
No doctests??  :-(

**Comment by PatOnTheBack** [comment]
@syedwaleedhyder you can find information about [writing doctests on Python's website](https://docs.python.org/3/library/doctest.html).

**Comment by syedwaleedhyder** [comment]
Doctests added :) @cclauss @PatOnTheBack

**Comment by PatOnTheBack** [comment]
@syedwaleedhyder why are you testing `decimal_to_hexadecimal`?

**Comment by syedwaleedhyder** [comment]
@PatOnTheBack Corrected.

**Comment by syedwaleedhyder** [comment]
@PatOnTheBack How much time LGTM takes?

**Comment by syedwaleedhyder** [comment]
@cclauss Resolved. All test cases pass now.

File: conversions/decimal_to_binary.py [hunk 1]
```diff
@@ -35,9 +34,9 @@ def decimal_to_binary(num):
         True
     """
     assert type(num) in (int, float) and num == int(num)
-
+    num = int(num)
```
Comment (cclauss) [inline conversions/decimal_to_binary.py]: This fails...
```
>>> decimal_to_binary(2.0) == bin(2.0)
True
```

Comment (syedwaleedhyder) [reply conversions/decimal_to_binary.py]: @cclauss bin(2.0) raises exception. And my code also raises exception. But the test still fails. Any idea how to resolve this issue? Can I just hard code the error in the doctests?
`**********************************************************************
File "decimal_to_binary.py", line 34, in __main__.decimal_to_binary
Failed example:
    decimal_to_binary(2.0) == bin(2.0) # doctest: +ELLIPSIS
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.6/doctest.py", line 1330, in __run
        compileflags, 1), test.globs)
      File "<doctest __main__.decimal_to_binary[9]>", line 1, in <module>
        decimal_to_binary(2.0) == bin(2.0) # doctest: +ELLIPSIS
      File "decimal_to_binary.py", line 38, in decimal_to_binary
        raise TypeError("'float' object cannot be interpreted as an integer")
    TypeError: 'float' object cannot be interpreted as an integer
**********************************************************************
`

**Comment by cclauss** [comment]
__# doctest: +ELLIPSIS__ as is done at https://github.com/TheAlgorithms/Python/blob/4a5589f4fcb71fc6101132bad44981693ef65d2c/project_euler/problem_10/sol3.py#L31-L38

**Comment by cclauss** [comment]
The exceptions do not have to be en exact match as long as the exception type and exception string are the same.

**Comment by cclauss** [comment]
Are we ready to merge?

**Comment by syedwaleedhyder** [comment]
I think so. Still have a final look @cclauss .

**Comment by syedwaleedhyder** [comment]
`Trying:
    decimal_to_binary(0)
Expecting:
    '0b0'
ok
Trying:
    decimal_to_binary(2)
Expecting:
    '0b10'
ok
Trying:
    decimal_to_binary(7)
Expecting:
    '0b111'
ok
Trying:
    decimal_to_binary(35)
Expecting:
    '0b100011'
ok
Trying:
    decimal_to_binary(-2)
Expecting:
    '-0b10'
ok
Trying:
    decimal_to_binary(16.16) # doctest: +ELLIPSIS
Expecting:
    Traceback (most recent call last):
    ...
    TypeError: 'float' object cannot be interpreted as an integer
ok
Trying:
    decimal_to_binary('0xfffff') # doctest: +ELLIPSIS
Expecting:
    Traceback (most recent call last):
    ...
    TypeError: 'str' object cannot be interpreted as an integer
ok
1 items had no tests:
    decimal_to_binary
1 items passed all tests:
   7 tests in decimal_to_binary.decimal_to_binary
7 tests in 2 items.
7 passed and 0 failed.
Test passed.
`

**Comment by cclauss** [comment]
Thanks for your patience and persistence!

**Comment by syedwaleedhyder** [comment]
Liked to work with you. Thanks. @cclauss

**Pull request merged**